src/main/java/hng_java_boilerplate/product/service/ProductServiceImpl.java

```diff
@@ -1,19 +1,22 @@
 package hng_java_boilerplate.product.service;
 
 import hng_java_boilerplate.product.dto.ErrorDTO;
+import hng_java_boilerplate.product.dto.ProductDTO;
 import hng_java_boilerplate.product.entity.Product;
 import hng_java_boilerplate.product.exceptions.ValidationError;
 import hng_java_boilerplate.product.repository.ProductRepository;
+import hng_java_boilerplate.user.repository.UserRepository;
+import hng_java_boilerplate.user.entity.User;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
-import java.util.List;
 
 @Service
 
 public class ProductServiceImpl implements ProductService{
 
-    private ProductRepository productRepository;
+    private final ProductRepository productRepository;
+    private UserRepository userRepository;
```

Comment:
I can seem to find where you did dependency injection for the user repository.

Reply:
But that's the dependency injection for the user repository

Reply:
Fixed this now